services/users.js

```diff
@@ -644,4 +651,36 @@ module.exports = class UsersService {
     return UserModel.find({status: 'PENDING'});
   }
 
+  /**
+   * Gives the user the ability to edit their username.
+   * @param  {String} id the id of the user to be toggled.
+   * @param  {Boolean} canEditName sets whether the user can edit their name.
+   * @return {Promise}
+   */
+  static toggleNameEdit(id, canEditName) {
+    return UserModel.update({id}, {
+      $set: {canEditName}
+    });
+  }
+
+  /**
+   * Gives the user the ability to edit their username.
+   * @param  {String} id the id of the user to be enabled.
+   * @param  {String}  displayName The new displayname for the user.
+   * @return {Promise}
+   */
+  static editName(id, displayName) {
+    return UserModel.findOne({id})
+      .then((user) => {
+        return user.canEditName ?
+        UserModel.update({id}, {
+          $set: {
+            displayName: displayName.toLowerCase(),
+            canEditName: false,
+            status: 'PENDING'
+          }
+        })
```

Comment:
Same as above:

```js
return UserModel.update({
  id,
  canEditName: true
}, {
  $set: {
    displayName: displayName.toLowerCase(),
    canEditName: false,
    status: 'PENDING'
  }
});
```

This will not return an error if they aren't allowed to do this, but I think this is a safer tradeoff.